payments/models.py

```diff
@@ -154,3 +169,345 @@ def __str__(self):
+                )
+
+            # If the lease is being changed
+            if old_instance.lease and self.lease != old_instance.lease:
+                raise ValidationError(
+                    _("Cannot change the lease associated to this order")
+                )
+
+    def is_full_season(self):
+        return (
+            self.lease.start_date == calculate_berth_lease_start_date()
+            and self.lease.end_date == calculate_berth_lease_end_date()
+        )
+
+    def _create_order_lines(self, section):
+        for service in ProductServiceType.FIXED_SERVICES():
+            # If the section has the service (prop) and it's valid (True)
+            if hasattr(section, service.name.lower()) and getattr(
+                section, service.name.lower()
+            ):
+                # Retrieve the product for that service
+                product = AdditionalProduct.objects.get(
+                    service=service, period=PeriodType.SEASON
+                )
+
+                OrderLine.objects.create(
+                    order=self, product=product,
+                )
+
+    def save(self, *args, **kwargs):  # noqa: C901  is too complex
+        self.full_clean()
+
+        # If the product is being added from the admin (only the ID is passed)
+        if self._product_object_id and not self.product:
+            # Try to get a BerthProduct (BP)
+            product = BerthProduct.objects.filter(id=self._product_object_id).first()
+            # If the BP was not found, try getting a WinterStorageProduct
+            product = (
+                product
+                if product
+                else WinterStorageProduct.objects.filter(
+                    id=self._product_object_id
+                ).first()
+            )
+            # If for some reason neither was found (shouldn't be the case), raise an error
+            if not product:
+                raise ValidationError(_("The product passed is not valid"))
+
+            self.product = product
+
+        # If the lease is being added from the admin (only the ID is passed)
+        if self._lease_object_id and not self.lease:
+            # Try to get a BerthLease (BL)
+            lease = BerthLease.objects.filter(id=self._lease_object_id).first()
+            # If the BL was not found, try getting a WinterStorageLease
+            lease = (
+                lease
+                if lease
+                else WinterStorageLease.objects.filter(id=self._lease_object_id).first()
+            )
+            # If for some reason neither was found, raise an error
+            if not lease:
+                raise ValidationError(_("The lease passed is not valid"))
+
+            self.lease = lease
+
+        creating = self._state.adding
+        if creating:
+            # If the product instance is being passed
+            # Price has to be assigned before saving but only if creating
+            if self.product:
```

Comment:
you can decrease one `if` by doing `if creating and self.product:` maybe

Reply:
Good catch! The structure was left from when there were more things inside the `if`, but now that there's not, it can be simplified. 😄